src/healthData/HealthDataManager.js

```diff
@@ -0,0 +1,180 @@
+            guid = uuid.v4();
+            PreferencesManager.setViewState("GUID", guid);
+        }
+
+        oneTimeHealthData.guid = guid;
+        oneTimeHealthData.snapshotTime = (new Date()).getTime();
+        oneTimeHealthData.os = brackets.platform;
+        oneTimeHealthData.osVersion = HealthDataUtils.getOSVersion();
+        oneTimeHealthData.osLanguage = brackets.app.language;
+        oneTimeHealthData.bracketsLanguage = brackets.getLocale();
+        oneTimeHealthData.bracketsVersion = brackets.metadata.version;
+
+        HealthDataUtils.getInstalledExtensions()
+            .done(function (userInstalledExtensions) {
+                oneTimeHealthData.installedExtensions = userInstalledExtensions;
+            })
+            .always(function () {
+                return result.resolve(oneTimeHealthData);
+            });
+
+        return result.promise();
+    }
+
+    /**
+     * Send data to the server
+     */
+    function sendHealthDataToServer() {
+        var result = new $.Deferred();
+        
+        getHealthData().done(function (jsonData) {
+            var url;
+            if (params.get("testEnvironment")) {
+                url = brackets.config.healthDataTestServerURL;
+            } else {
+                url = brackets.config.healthDataServerURL;
+            }
+
+            $.ajax({
+                url: url,
+                type: "POST",
+                data: jsonData,
+                dataType: "text",
+                contentType: "text/plain"
+            })
+                .done(function () {
+                    result.resolve();
+                })
+                .fail(function (jqXHR, status, errorThrown) {
+                    console.error("Error in sending health data. Response : " + jqXHR.responseText + ". Status : " + status + ". Error : " + errorThrown);
+                    result.reject();
+                })
+                .always(function () {
+                    PreferencesManager.setViewState("lastTimeSendHealthData", (new Date()).getTime());
+                });
+        })
+            .fail(function () {
+                result.reject();
+            });
+
+        return result.promise();
+    }
+
+    /*
+     * Check if health data is to be send to the server. If user has enable the tracking, health data will be send for every 24 hours.
+     * If 24 hours or more than that has been passed, then send health data to the server
+     */
+    function _checkHealthDataExport() {
+        var result = new $.Deferred();
+        var isHDTracking = prefs.get("healthDataTracking");
+        window.clearTimeout(timeoutVar);
+
+        if (isHDTracking) {
+
+            var lastTimeSend = PreferencesManager.getViewState("lastTimeSendHealthData"),
+                currentTime = (new Date()).getTime();
+
+            if (!lastTimeSend) {
+                var randomTime = Math.floor(Math.random() * ONE_DAY);
+                lastTimeSend = currentTime + randomTime;
```

Comment:
@MarcelGerber We don't want to immediately send data on first launch because the user needs time to opt out if they want to.

@prafulVaishnav Remember preferences persist on upgrades, so when the big upgrade spike happens the vast majority of users who are upgrading won't hit this codepath anyway.  It's only really an issue if some sort of viral PR thing happened, with many _brand new_ users coming in at once -- and even then I'd expect new-user first launches to be distributed in time enough that randomization may not be needed.

Since it adds complexity, we should think about whether this wrinkle in the logic is really needed.  If you still feel like it's important, please add a comment though explaining that the purpose is to distribute server load (and pointing out that the _minimum_ time before the first ping is 24 hours, due to the `+ ONE_DAY` in the `else` case below).